src/components/MainButton.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+  <button :class="buttonClasses">{{ buttonText }}</button>
+</template>
+
+<script>
+export default {
+  name: "MainButton",
+  props: {
+    buttonText: {
+      type: String,
+      default: "Click Me!",
+    },
+    buttonType: {
+      type: String,
+      default: "primary",
+    },
+  },
+  computed: {
+    buttonClasses() {
+      return [
+        "button",
+        { button__primary: this.buttonType === "primary" },
+        { button__secondary: this.buttonType === "secondary" },
```

Comment:
according to BEM convention, `primary` and `secondary` should probably be "modifiers" (not "elements")
http://getbem.com/naming/